### PR TITLE
Cl/skeleton use case

### PR DESCRIPTION
**Pull request by AsiaWi**
Created use cases and associated constructs in application tier.

### CODE REVIEW

**Review by spanersoraferty — APPROVED** [review]

File: Dfe.Data.SearchPrototype/SearchForEstablishments/SearchContext.cs [hunk 1]
```diff
@@ -16,6 +16,8 @@ public sealed class SearchContext
         public string TargetCollection { get; }
 
         /// <summary>
+        /// The following arguments are passed via the constructor and are not changeable
+        /// once an instance is created, this ensures we preserve immutability.
```
Comment (CathLass) [inline Dfe.Data.SearchPrototype/SearchForEstablishments/SearchContext.cs:20]: Instead, can the summary be "Encapsulates the data necessary to perform the search"

File: Dfe.Data.SearchPrototype/Tests/SearchForEstablishments/SearchByKeywordUseCaseTests.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        ISearchServiceAdapter searchServiceAdapter =
+            SearchServiceAdapterTestDouble.MockFor(
+                EstablishmentResultsTestDouble.Create());
+
+        IMapper<EstablishmentResults, SearchByKeywordResponse> mapper = new ResultsToResponseMapper();
+        _useCase = new(searchServiceAdapter, mapper);
+    }
+
+    [Fact]
+    public async Task UseCase_ValidRequest_ReturnsValidResponse()
```
Comment (CathLass) [inline Dfe.Data.SearchPrototype/Tests/SearchForEstablishments/SearchByKeywordUseCaseTests.cs]: I'm up for renaming this test UseCase_ValidRequest_ReturnsResponse() at the minute, as there's nothing to check we have a valid response (other than it doesn't throw). Unless we want to add asserts to check the response has results and call the test UseCase_ValidRequestWithResults_ReturnsResponseWithResults() or something 🤔

**Review by CathLass — COMMENTED** [review]
I've added some suggestions for bits and bobs before merging but I'm not wedded to them

File: Dfe.Data.SearchPrototype/SearchForEstablishments/SearchContext.cs (hunk 1 above)
Comment (spanersoraferty) [reply Dfe.Data.SearchPrototype/SearchForEstablishments/SearchContext.cs:20]: I think it's useful to state this is immutable IMO, but happy to debate ;)

File: Dfe.Data.SearchPrototype/Tests/SearchForEstablishments/SearchByKeywordUseCaseTests.cs (hunk 2 above)
Comment (spanersoraferty) [reply Dfe.Data.SearchPrototype/Tests/SearchForEstablishments/SearchByKeywordUseCaseTests.cs]: Yip, agreed

**Pull request merged**